lib/livebook_web/live/session_live/section_component.ex

```diff
@@ -77,11 +77,19 @@ defmodule LivebookWeb.SessionLive.SectionComponent do
             {if @section_view.has_children?,
                do: [class: "tooltip left", data_tooltip: "Cannot delete this section because\nother sections branch from it"],
                else: [class: "tooltip top", data_tooltip: "Delete"]}>
-            <%= live_patch to: Routes.session_path(@socket, :delete_section, @session_id, @section_view.id),
-                  class: "icon-button #{if @section_view.has_children?, do: "disabled"}",
-                  aria_label: "delete section",
-                  role: "button" do %>
-              <.remix_icon icon="delete-bin-6-line" class="text-xl" />
+            <%= if @section_view.cell_views == [] or Enum.all?(@section_view.cell_views, & &1.empty?) do %>
+               <button class="icon-button #{if @section_view.has_children?, do: 'disabled'}"
+                 aria-label="delete section"
+                 phx-click="delete_empty_section"
```

Comment:
I think it would be a bit cleaner to always have a button with `phx-click="delete_section"` and then in `handle_event` we make the check and either delete the section straight away or `push_patch` to redirect to the modal. WDYT?

Reply:
Thanks Jonatan.

Yeah good idea. I've made that change in https://github.com/livebook-dev/livebook/pull/829/commits/7e6d2fa21e8a7f77f320b141e835d1e5e88a5513 -- I do prefer having a single callback where all of the prompt checking logic is contained, but it has made the callback a little messier than I'd like. Two things:

1. WDYT about extracting the `section_view` here? It felt kinda gross. I did also try extracting the `notebook` and then using `Notebook.fetch_section(section_id)`, but the original cells structs don't include the `empty?` attribute and I couldn't figure out how to check if the content is empty via the `source` (it was always `nil`?). 
2. I also thought about having a `Notebook.is_section_empty?/2` but calling it with a `section_id` rather than a `section` itself also didn't feel right (and once I'd fetched the section, having an additional function seemed superfluous).

Let me know what you think!

Reply:
I've just realized `empty?` reflects the initial state on initial render and is not updated when cell contents change. We don't keep/update the cell text in the LV to save on memory (as you noticed cell `source` is `nil`, because we explicitly set it after the initial render). Other than that, empty cells may have editor history, so I think in that case it's actually better to show the confirm.

The `*_view` data is a stripped down version of the whole data that we use in `render` as an optimisation. In the callbacks you can rather use `Notebook.fetch_section(socket.private.data.notebook, section_id)`, then only check if `section.cells == []` :)

Reply:
> Other than that, empty cells may have editor history, so I think in that case it's actually better to show the confirm.

Thanks for explaining that. Makes sense!

I've pushed another change to use `Notebook.fetch_section` and `section.cells` -- wasn't sure if I needed to handle the case where a section might not exist but it seems we match explicitly in other places across the live view already so I copied that.

Reply:
We want to handle `:error` too, by the time we handle the deletion the section may have already been deleted by another user.

> it seems we match explicitly in other places across the live view already

Only in those places where we are sure the session should exist, in event handlers we are not sure since the session id comes from the user at any point.

Reply:
Good point. Handled that case. Let me know if you have any further style suggestions.

Reply:
Perfect!